### PR TITLE
Deploy to Istio 1.0

### PR DESCRIPTION
**Pull request by jonhawkes**
Signed-off-by: Jon Hawkes <jon.hawkes@uk.ibm.com>

### CODE REVIEW

File: samples/istio/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+## MicroProfile Istio Sample
+This sample is intended to show the interaction between Istio and the various MicroProfile specifications.
+
+### Requirements
+* [Git](https://git-scm.com/)
+* [Docker](https://www.docker.com/)
+* [Maven](https://maven.apache.org/install.html)
+* [Java 8]: Any compliant JDK should work.
+  * [Java 8 JDK from Oracle](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
```
Comment (pilhuhn) [inline samples/istio/README.md]: Remove mentioning specific JDKs/JREs

Comment (Emily-Jiang) [inline samples/istio/README.md]: On today's call, we agreed the subsequent list is not needed.

File: samples/istio/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+Follow the instructions in the READMEs there to build the sample services with maven.
+
+### Push
+
+The services should be packaged within your docker development environment. Each service has a Dockerfile which can be used to assemble its docker image. Each image should be tagged such that it can be subsequently pushed to your docker repository. An example tag would be <your docker id>/servicea.
+
+Each service image needs to be published to a repository so it will be available to run in your cluster. Assuming you will be using Docker Hub and following on with the example tag for service-a:
+1.  docker login
+3.  docker push <your docker id>/servicea
```
Comment (pilhuhn) [inline samples/istio/README.md]: This needs changing according to the naming strategy.
Same below for serviceb

File: samples/istio/deployments.yaml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: servicea
+    spec:
+      containers:
+      - name: servicea
+        image: SERVICE_A_TAG:latest
```
Comment (pilhuhn) [inline samples/istio/deployments.yaml]: Also needs changed to the naming strategy (?)

File: samples/istio/deployments.yaml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: serviceb
+    spec:
+      containers:
+      - name: serviceb
+        image: SERVICE_B_TAG:latest
```
Comment (pilhuhn) [inline samples/istio/deployments.yaml]: See naming strategy

File: samples/istio/deployments.yaml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  name: serviceb-deployment
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: serviceb
+    spec:
+      containers:
+      - name: serviceb
```
Comment (Emily-Jiang) [inline samples/istio/deployments.yaml]: we need to unify the image name and refer to the docker image created in the sample repo service-mesh-service-b

File: samples/istio/deployments.yaml [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      labels:
+        app: servicea
+    spec:
+      containers:
+      - name: servicea
+        image: SERVICE_A_TAG:latest
+        imagePullPolicy: Always
+        ports:
+        - containerPort: 8080
+        envFrom:
```
Comment (pilhuhn) [inline samples/istio/deployments.yaml]: @Emily-Jiang This is what we were talking about on the call. As it is a `envFrom/configMapRef` it works and exposes the config map as env variables, so all is good.

Comment (Emily-Jiang) [reply samples/istio/deployments.yaml]: cool. thanks for your confirmation!

Comment (Emily-Jiang) [reply samples/istio/deployments.yaml]: @pilhuhn please can you approve the other config PR?

File: samples/istio/services.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+metadata:
+  name: serviceb-service
+  labels:
+    app: serviceb
+spec:
+  ports:
+  - port: 8080
+    name: http
+  selector:
+    app: serviceb
```
Comment (Emily-Jiang) [inline samples/istio/services.yaml:25]: This needs to correspond to the new docker image name.

File: samples/istio/services.yaml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+metadata:
+  name: servicea-service
+  labels:
+    app: servicea
+spec:
+  ports:
+  - port: 8080
+    name: http
+  selector:
+    app: servicea
```
Comment (Emily-Jiang) [inline samples/istio/services.yaml:12]: This needs to correspond to the new docker image name.

File: samples/istio/deployment-template.yaml [hunk 9]
```diff
@@ -0,0 +1,28 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: DEPLOYMENT-deployment
```
Comment (pilhuhn) [inline samples/istio/deployment-template.yaml:4]: Add server-kind (liberty/thorntail) as `version: <kind>` label here

Comment (Emily-Jiang) [reply samples/istio/deployment-template.yaml:4]: servicea-liberty-deployment
servicea-thorntail-deployment

File: samples/istio/deployment-template.yaml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
```
Comment (Emily-Jiang) [inline samples/istio/deployment-template.yaml:9]: labels:
        app: servicea
        runtime: thorntail
        mp-version: mp-1.3
        version: thorntail/mp-1.3

**Comment by jonhawkes** [comment]
This will be superceded.

**Pull request closed without merging**